extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/WithSession.java

```diff
@@ -16,8 +16,8 @@
  * triggered, then this session is reused. Otherwise, a new session is opened and eventually closed when the
  * {@link io.smallrye.mutiny.Uni} completes.
  * <p>
- * A method annotated with this annotation must return either {@link io.smallrye.mutiny.Uni}. If declared on a class then all
- * methods that are intercepted must return {@link io.smallrye.mutiny.Uni}.
+ * A method annotated with this annotation must return {@link io.smallrye.mutiny.Uni}. If declared on a class then all methods
+ * that return {@link io.smallrye.mutiny.Uni} are considered; all other methods are ignored.
```

Comment:
Do we really want this? I guess it's an inconvenience to have to annotate multiple methods if necessary, but isn't it better than having a method ignored because the user forgot that it must return a Uni?

Reply:
Well, it depends. I agree that the current behavior is a bit more correct. 

But  sometimes you need transaction/session for 80% of the methods but not for few package-private or protected methods, usually some utils inherited from a superclass. And in this case it's very inconvenient to annotate each method.

Reply:
To me is the difference between adding a couple of annotations and having to debug an issue that's hard to spot.

Reply:
Anyway, I don't have a strong opinion, we could also add a property in the annotation to enable/disable strict validation